diff --git a/node_modules/vue-resize/dist/vue-resize.esm.js b/node_modules/vue-resize/dist/vue-resize.esm.js
index b394524..395ba7d 100644
--- a/node_modules/vue-resize/dist/vue-resize.esm.js
+++ b/node_modules/vue-resize/dist/vue-resize.esm.js
@@ -55,7 +55,9 @@ var ResizeObserver = { render: function render() {
 				if (!isIE && this._resizeObject.contentDocument) {
 					this._resizeObject.contentDocument.defaultView.removeEventListener('resize', this.compareAndNotify);
 				}
-				delete this._resizeObject.onload;
+				this.$el.removeChild(this._resizeObject);
+				this._resizeObject.onload = null;
+				this._resizeObject = null;
 			}
 		}
 	},
